challenges/3/YOUR-CODE-GOES-HERE/mehrannaseri/app/Http/Controllers/TravelSpotController.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Enums\TravelStatus;
+use App\Exceptions\InvalidTravelStatusForThisActionException;
+use App\Exceptions\ProtectedSpotException;
+use App\Exceptions\SpotAlreadyPassedException;
+use App\Http\Requests\TravelSpotStoreRequest;
+use App\Http\Resources\TravelResource;
+use App\Models\Driver;
+use App\Models\Travel;
+use App\Models\TravelSpot;
+use Carbon\Carbon;
+
+class TravelSpotController extends Controller
+{
+	public function arrived(Travel $travel, TravelSpot $spot)
+	{
+        $driver = auth()->user();
+        if(! Driver::isDriver($driver))
+            return response()->json([], 403);
+
+        if($travel->status != TravelStatus::RUNNING)
+            throw new InvalidTravelStatusForThisActionException();
+
+        if($travel->driverHasArrivedToOrigin())
+            throw new SpotAlreadyPassedException();
+
+        $origin = $travel->getOriginSpot();
+        $origin->update([
+            'arrived_at' => Carbon::now()
+        ]);
+
+        return response()->json(TravelResource::make($travel));
+	}
+
+	public function store(Travel $travel, TravelSpotStoreRequest $request)
+	{
+        $user = auth()->user();
+
+        if(Driver::isDriver($user))
+            return response()->json([], 403);
+
+        if($travel->allSpotsPassed())
```

Comment:
So at end of travel, the passenger wants to go to another location, it's wrong logically?